cmake/MOAB_PullAndMake.cmake

```diff
@@ -0,0 +1,93 @@
+# This Macro sets up the download and build of MOAB using ExternalProject
+# few tweak are done in src/dagmc/CMakeLists.txt and src/PyNE/CMakeists.txt 
+# to make sure that MOAB is built before DAGMC.
+MACRO (moab_pull_make)
+  MESSAGE(STATUS "MOAB will be downloaded and built")
+  
+  # Ensure STATIC LIBS Build
+  IF(DAGMC_BUILD_STATIC_LIBS)
+    MESSAGE(FATAL_ERROR "PULL_INSTALL_MOAB is ONLY compatible with shared libraries.")
+  endIF()
+  
+  # Ensure MOAB_VERSION
+  IF(NOT MOAB_VERSION)
+    SET(MOAB_VERSION "5.5.1")
+  endIF()
```

Comment:
It might be useful to do this at the top level so this pinned version string is more obvious if/when it needs to be changed.